crates/router/src/core/payments.rs

```diff
@@ -3713,8 +3721,53 @@ pub async fn decide_multiplex_connector_for_normal_or_recurring_payment<F: Clone
                     .as_ref()
                     .ok_or(errors::ApiErrorResponse::InternalServerError)
                     .attach_printable("Failed to find the merchant connector id")?;
+                if is_network_token_with_transaction_id_flow(
+                    state,
+                    connector_data.connector_name,
+                    is_connector_agnostic_mit_enabled,
+                    is_network_tokenization_enabled,
+                    payment_method_info,
+                ) {
+                    if let Ok((token_exp_month, token_exp_year)) =
+                        do_status_check_for_network_token(state, key_store, payment_method_info)
+                            .await
+                    {
+                        logger::info!(
+                            "using network_tokenization with network_transaction_id for MIT flow"
+                        );
```

Comment:
Should we consider also logging the identifiers of any entities involved, say network transaction ID, payment method ID, etc., if it can help improve debugging?

Reply:
here we only decide the Mandate flow, hence the message only says which flow is used for that MIT